api/src/pcapi/routes/backoffice/venues/blueprint.py

```diff
@@ -321,24 +322,30 @@ def get_venue_bank_information(venue: offerers_models.Venue) -> serialization.Ve
 
 @venue_blueprint.route("/<int:venue_id>/stats", methods=["GET"])
 def get_stats(venue_id: int) -> utils.BackofficeResponse:
-    venue = (
-        offerers_models.Venue.query.filter(offerers_models.Venue.id == venue_id)
-        .options(
-            sa.orm.joinedload(offerers_models.Venue.pricing_point_links).joinedload(
-                offerers_models.VenuePricingPointLink.pricingPoint
-            ),
-            sa.orm.joinedload(offerers_models.Venue.reimbursement_point_links)
-            .joinedload(offerers_models.VenueReimbursementPointLink.reimbursementPoint)
-            .load_only(offerers_models.Venue.name)
-            .joinedload(offerers_models.Venue.bankInformation)
-            .load_only(finance_models.BankInformation.bic, finance_models.BankInformation.iban),
-            sa.orm.joinedload(offerers_models.Venue.bankInformation).load_only(
-                finance_models.BankInformation.bic, finance_models.BankInformation.iban
-            ),
-        )
-        .one_or_none()
+    venue_query = offerers_models.Venue.query.filter(offerers_models.Venue.id == venue_id).options(
+        sa.orm.joinedload(offerers_models.Venue.pricing_point_links).joinedload(
+            offerers_models.VenuePricingPointLink.pricingPoint
+        ),
+        sa.orm.joinedload(offerers_models.Venue.reimbursement_point_links)
+        .joinedload(offerers_models.VenueReimbursementPointLink.reimbursementPoint)
+        .load_only(offerers_models.Venue.name)
+        .joinedload(offerers_models.Venue.bankInformation)
+        .load_only(finance_models.BankInformation.bic, finance_models.BankInformation.iban),
+        sa.orm.joinedload(offerers_models.Venue.bankInformation).load_only(
+            finance_models.BankInformation.bic, finance_models.BankInformation.iban
+        ),
     )
 
+    if FeatureToggle.WIP_ENABLE_NEW_BANK_DETAILS_JOURNEY.is_active():
+        venue_query = venue_query.options(
+            sa.orm.joinedload(offerers_models.Venue.bankAccountLinks)
```

Comment:
Est-ce que c’est vraiment utiles de récupérer tous les VenueBankAccountLink ? De toute manière on ne va afficher que celui qui est d’actualité ou None via `current_bank_account_link` donc autant ne pas overfetch non ?

Reply:
`current_bank_account_link` parcourt `bankAccountLinks` donc si on ne l'ORM ne les a pas récupérés, alors une requête N+1 va le faire au moment de l'appel à `current_bank_account_link`.

Reply:
Je ne parlais pas de ne pas les fetcher, juste de filtrer sur la valeur du timespan

Reply:
Comment ferais-tu cela dans le code, avec le joinedload, pour que ça ne cause pas une requête N+1 dans `Venue.current_bank_account_link` ?

Reply:
Dans la même logique que dans le repository finance dans `get_bank_account_with_current_venues_link` ?

Reply:
Je viens de pousser un second commit qui fait différemment, avec une sous-requête générant les données dont on a besoin pour le compte bancaire, on peut alors oublier `Venue.current_bank_account_link`.

Si les revues préfèrent majoritairement la première implémentation, je garde seulement le premier commit.
Si la préférence va à la seconde (moins de lignes SQL, mais sans l'ORM pour ces infos), alors je squash les deux commits.

Reply:
Je ne sais pas si c’est nécessaire de faire une subquery et de faire faire la mise en forme des données à sqlalchemy mais je parlais juste de faire ça:

```suggestion
        venue_query = venue_query.options(
            outerjoin(offerers_models.VenueBankAccountLink, sqla.and_(Venue.id==offerers_models.VenueBankAccountLink.venueId, offerers_models.VenueBankAccountLink.timespan.contains(utc.now())),
            .outerjoin(offerers_models.VenueBankAccountLink.bankAccount)
            .load_only(finance_models.BankAccount.id, finance_models.BankAccount.label)
        )
```

Reply:
Ok, je comprends, avec un `outerjoin` conditionnel et `contains_eager` on lui en donne un `joinedload` partiel mais suffisant pour les données dont on a besoin. Je viens donc de pousser une nouvelle version (un commit qui remplace les précédents). Merci !

Reply:
je t’en prie !